src/main/java/edu/psu/swe/commons/jaxrs/hateoas/interceptor/HateoasWriteInterceptor.java

```diff
@@ -147,9 +147,11 @@ private void addAtomLink(Link linkAnnotation, Object object) {
       }
     }
 
-    atomLink.setHyperlink(uriInfo.getBaseUri() + path);
-
-    instance.getLinks().add(atomLink);
+    if (path != null && !path.trim().isEmpty()) { 
+      atomLink.setHyperlink(uriInfo.getBaseUri() + path);
```

Comment:
This will indeed block the inclusion of null HATEOAS references but reviewing your change leads me to ask another question - Since RESTful tenets state that there should be one definitive URI for each resource, what happens with the ```uriInfo.getBaseUri()``` call in situations where there are multiple servers behind a load balancer?  This code should probably retrieve a system property and fall back to the value provided by the ```UriInfo``` class if it isn't available.